### ISSUE TITLE
logstash hanging on startup

### ISSUE THREAD

**Issue by aparajit0** [cisagov/Malcolm#235]
# 🐛 Summary #

Installed vanilla using quickstart guide in lubuntu 22.04.1. logstash repeats the fail loop below until ./scrips/stop is ran.

## To reproduce ##

1. user@lubuntu:~/Malcolm$ ./scripts/start

What did you expect to happen that didn't?

Malcolm status running

## Any helpful log output or screenshots ##

Paste the results here:

logstash_1            | 2023-02-05 19:48:22,012 INFO spawned: 'logstash' with pid 83146
logstash_1            |
logstash_1            | 2023-02-05 19:48:22,652 INFO success: logstash entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
logstash_1            | 2023/02/05 19:48:22 Setting 'pipeline.batch.delay' from environment.
logstash_1            | 2023/02/05 19:48:22 Setting 'pipeline.batch.size' from environment.
logstash_1            | 2023/02/05 19:48:22 Setting 'pipeline.workers' from environment.
logstash_1            | Using bundled JDK: /usr/share/logstash/jdk
logstash_1            | Error: Unable to initialize main class org.logstash.Logstash
logstash_1            | Caused by: java.lang.NoClassDefFoundError: org/jruby/exceptions/RaiseException
logstash_1            | 2023-02-05 19:48:23,832 WARN exited: logstash (exit status 1; not expected)

**Comment by mmguero** [comment]
Hmmm, with this amount of information I'm not able to hazard a guess. This happens repeatedly even after using ./scripts/wipe and then restarting with ./scripts/start?

What are the resources (specifically available RAM) of the host? And how much memory allocated for Logstash and OpenSearch (`grep -Po "\w+_JAVA_OPTS.+Xm[sx]\w+" docker-compose*` will give you those figures)? Usually containers failing to stay started are dealing with limited resources.

Beyond that, maybe you can get it started up and into this state and then run `docker-compose logs > output.txt` and attach the contents of that file to this issue and I can try to make a better guess.

**Comment by aparajit0** [comment]
I apologize for my delayed response to your prompt support. I have been
flooded with work since coming back from S4 and your initial email was
buried in my personal email.

Your suspicion of ram allocation for logstash and opensearch was spot on.
While I followed the quickstart guide, I didn't requisition half of my
available ram allocated to the system for opensearch and only provided the
minimum ram required for logstash. I have rectified and remediated it with
success.

Thank you for reminding me to read the fantastic documentation your team
has created in greater detail by offering me the most relevant bits in an
email for my corrective action.

-Brett
